src/Prismic/Exception/RequestFailureException.php

```diff
@@ -18,23 +20,34 @@ class RequestFailureException extends RuntimeException
 
     /**
      * Factory to return a Prismic Exception wrapping a Guzzle Exception
+     *
+     * @param GuzzleException $e
+     *
+     * @return self
      */
-    public static function fromGuzzleException(GuzzleException $e) : self
+    public static function fromGuzzleException(GuzzleException $e): self
     {
-        if ($e instanceof RequestException) {
-            return static::fromGuzzleRequestException($e);
+        if ($e instanceof RequestException || $e instanceof ConnectException) {
+            return static::fromGuzzleRequestOrConnectException($e);
         }
+
         $exception = new static('Api Request Failed', 500, $e);
         $exception->guzzleException = $e;
         return $exception;
     }
 
+
     /**
-     * Factory to wrap a Guzzle Request Exception when we should have access to a request and a response
+     * Factory to wrap a Guzzle Request or Connect Exception when we should have access
+     * to a request and a optionally response
+     *
+     * @param RequestException|ConnectException $e
+     *
+     * @return self
      */
-    protected static function fromGuzzleRequestException(RequestException $e) : self
+    protected static function fromGuzzleRequestOrConnectException(GuzzleException $e): self
```

Comment:
@gsteel Is this renaming okay? In theory, someone could extend this class and call the function...

Reply:
I had the Idea to keep the old method in place (with a deprecation notice) and letting it call the new function. Then it can be removed in the next major, correct?

Reply:
Sorry, I see the type hint has also changed - Yes a new method would be better and `/** @deprecated **/` for the old method. I'd make the new method private if possible and still mark the class with `/** @final */`

Reply:
Now I have the commit ready with the change. I think everything should be good to merge and ship as `v5.3.0` right?